temporalio/worker/_workflow_instance.py

```diff
@@ -1845,6 +1856,14 @@ async def _run_top_level_workflow_function(self, coro: Awaitable[None]) -> None:
                 err
             ):
                 self._add_command().cancel_workflow_execution.SetInParent()
+                self._warn_if_unfinished_handlers()
```

Comment:
I don't think we should warn before sending the cancellation. A user should have the opportunity to receive and react to cancellation without warning.